### PR TITLE
GSoC Week 2 PR: Created the wrapper class with basic functionality

### PR DESCRIPTION
**Pull request by 3V3RYONE**
Created a initial wrapper class for HTTP-Trace. Verified that the initialization message is shown when an object of the wrapper class is created.  (Task 1 of Week 2)  
  
Week 2 Tasks completed.

### CODE REVIEW

**Comment by 3V3RYONE** [comment]
Established and verified the connection betn wrapper class and client and server. This completes Task 1 of Week 2.  
  
Output showing the two methods of wrapper class being called when HTTP-Tracing is put to action.  
  
output_week2_task1

File: lib/rex/proto/http/server.rb [hunk 1]
```diff
@@ -40,6 +40,10 @@ def send_response(response)
     # Add any other standard response headers.
     server.add_response_headers(response)
 
+    # Use HTTP Trace for server
+    http_trace_object = Rex::Proto::Http::HttpTrace.new
+    http_trace_object.use_http_trace_response(response, 'red/blue')
+
```
Comment (gwillcox-r7) [inline lib/rex/proto/http/server.rb]: This is only logging responses, what about requests that come in to the HTTP server? Those are also important to log and show to the end user.

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]
Missing logging responses from the server. Also would have liked to have seen some initial attempts to display the data vs just displaying static messages. Doesn't have to be perfect but shows at least your looking into how to render the data.

**Comment by 3V3RYONE** [comment]
With the new commits, I allowed logging of requests at server side. Also, the definition of methods and constructor of the class have been improved, i.e. now they accept the datastore options for HTTP-Trace, and if the option is set, the tracing is enabled.  
  
updated_output  
  
This implements all 3 Tasks of Week 2.

**Review by gwillcox-r7 — APPROVED** [review]
LGTM, go for it :)

**Pull request merged**